CMakeLists.txt

```diff
@@ -0,0 +1,152 @@
+cmake_minimum_required(VERSION 2.8.12)
+
+project(cctz)
+
+set(CMAKE_MODULE_PATH
+  ${PROJECT_SOURCE_DIR}/cmake
+  ${PROJECT_SOURCE_DIR}/cmake/modules
+  ${CMAKE_MODULE_PATH})
+
+include(CheckCXXCompilerFlag)
+include(CTest)
+include(GNUInstallDirs)
+
+option(BUILD_EXAMPLES "Whether or not to build examples" ON)
+
+if (CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
+  add_compile_options(-Wall)
+endif()
+
+# if a specific standard is required, this can be set from the command line:
+#     cmake -DCMAKE_CXX_STANDARD=[11|14|17]
+if (NOT CMAKE_CXX_STANDARD)
+  # C++11 is the minimum version required
+  set(CMAKE_CXX_STANDARD 11)
+endif()
+
+# CMAKE_CXX_STANDARD is only supported since CMake >= 3.1
+if (CMAKE_VERSION VERSION_LESS "3.1")
+  if (CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
+    add_compile_options(-std=c++${CMAKE_CXX_STANDARD})
+  endif ()
+endif()
+
+# Setup the flag to enable colored diagnostics if any.
```

Comment:
It seems strange to me to include color diagnostics in a project like this, which is almost certainly going to have its own settings.

Reply:
I think I will remove this code as this is "not standard", and there is currently no way to enable or disable it.
I need to find a less intrusive way to do this.

However, I don't think it should be an issue in practice for projects including cctz in another projects, as `add_compile_options()` should not affect them.

Reply:
Just a follow up.

Using `-DCMAKE_CXX_ARGS=-fcolor-diagnostics` is a non-intrusive way to do this, so one can get colors with Ninja using the following `cmake` invocation:

    cmake -G Ninja -DCMAKE_CXX_ARGS=-fcolor-diagnostics ...

Can also be done with `CXXFLAGS=-fcolor-diagnostics cmake ...`.